frontend/src/components/visualizations/Visualization.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { Suspense, useEffect } from "react";
+import { LazyLoadErrorBoundary } from "../errors/LazyLoadErrorBoundary";
+import { useDispatch, useSelector } from "react-redux";
+import { RootState } from "../../types/data";
+import { fetchFileSummary } from "../../actions/file";
+import { getVizConfig as getVizConfigAction } from "../../actions/visualization";
+import { vizComponentDefinitions } from "../../visualization.config";
+
+type previewProps = {
+	fileId?: string;
+};
+export const Visualization = (props: previewProps) => {
+	const { fileId } = props;
+
+	const fileSummary = useSelector((state: RootState) => state.file.fileSummary);
+	const vizConfig = useSelector(
+		(state: RootState) => state.visualization.vizConfig
+	);
+
+	const dispatch = useDispatch();
+	const listFileSummary = (fileId: string | undefined) =>
+		dispatch(fetchFileSummary(fileId));
+
+	const getVizConfig = (resourceId: string | undefined) =>
+		dispatch(getVizConfigAction(resourceId));
+
+	useEffect(() => {
+		listFileSummary(fileId);
+		getVizConfig(fileId);
+	}, []);
+
+	return (
+		<>
+			{vizComponentDefinitions.map((vizComponentDefinition) => {
```

Comment:
This block might use a few more comments. Lots of steps going on that might not be clear.